### PR TITLE
[Market] DSM 2.0

### PR DESCRIPTION
**Pull request by mmyyrroonn**

### CODE REVIEW

File: cstrml/market/src/lib.rs [hunk 1]
```diff
@@ -71,6 +69,8 @@ pub trait WeightInfo {
 pub struct FileInfo<AccountId, Balance> {
     // The ordered file size, which declare by user
     pub file_size: u64,
+    // The size of used value in MPoW
+    pub used_size: u64,
```
Comment (badkk) [inline cstrml/market/src/lib.rs]: Change into storage_power

File: cstrml/market/src/lib.rs [hunk 2]
```diff
@@ -146,49 +132,52 @@ impl<T: Config> MarketInterface<<T as system::Config>::AccountId, BalanceOf<T>>
...
+                used_size = file_info.file_size;
             }
 
-            // 4. Update used_info
-            if is_counted {
-                used_size = Self::add_used_group(&mut used_info, anchor, cid); // need to add the used_size after the update
-            };
-
-            // 5. The first join the replicas and file become live(expired_on > calculated_at)
+            // 4. The first join the replicas and file become live(expired_on > calculated_at)
```
Comment (badkk) [inline cstrml/market/src/lib.rs]: `replic.len() == 1` -> `expired_on == 0`

File: cstrml/market/src/lib.rs [hunk 3]
```diff
@@ -197,24 +186,43 @@ impl<T: Config> MarketInterface<<T as system::Config>::AccountId, BalanceOf<T>>
     /// Accept id(who, anchor), cid and current block number
     /// Returns the real used size of this file
     fn delete_replica(who: &<T as system::Config>::AccountId, cid: &MerkleRoot, anchor: &SworkerAnchor) -> u64 {
+        let mut used_size: u64 = 0;
         // 1. Delete replica from file_info
-        if let Some((mut file_info, used_info)) = <Files<T>>::get(cid) {
+        if let Some(mut file_info) = <Files<T>>::get(cid) {
             let mut is_to_decreased = false;
+            let mut is_valid: Option<bool> = None;
```
Comment (badkk) [inline cstrml/market/src/lib.rs]: give some comments about `is_spower_counted`.

File: cstrml/market/src/lib.rs [hunk 4]
```diff
@@ -197,24 +186,43 @@ impl<T: Config> MarketInterface<<T as system::Config>::AccountId, BalanceOf<T>>
...
+            // 2. Return the original used size in wr
+            if let Some(is_valid) = is_valid {
+                if is_valid {
+                    used_size = file_info.used_size;
+                } else {
+                    used_size = file_info.file_size;
+                }
+            }
+
+            // 3. Decrease the reported_replica_count
```
Comment (badkk) [inline cstrml/market/src/lib.rs:214]: 217 needs to consider sub with same account different anchor?

File: cstrml/market/src/lib.rs [hunk 5]
```diff
@@ -315,6 +320,9 @@ decl_storage! {
         /// The file base fee for each storage order.
         pub FileBaseFee get(fn file_base_fee): BalanceOf<T> = Zero::zero();
 
+        /// The used size to become valid duration
+        pub ValidDuration get(fn valid_duration): BlockNumber = 1_296_000; // 3 months
```
Comment (badkk) [inline cstrml/market/src/lib.rs]: `SpowerReadyPeriod` or something?

**Review by badkk — CHANGES_REQUESTED** [review]

File: cstrml/market/src/lib.rs (hunk 1 above)
Comment (mmyyrroonn) [reply cstrml/market/src/lib.rs]: spower

**Review by badkk — APPROVED** [review]

**Comment by badkk** [comment]
Needs to migrate: `Used` -> `Spower` in swork module

**Pull request merged**